code/modules/wiremod/components/list/assoc_list_pick.dm

```diff
@@ -0,0 +1,75 @@
+/**
+ * # associative List pick component
+ *
+ * Allows user to select 1 entry from a list
+ */
+
+/obj/item/circuit_component/assoc_list_pick
+	display_name = "Associative List Pick"
+	desc = "A component that lets a user pick 1 element from an associative list. Returns the selected element."
+	category = "List"
+
+	var/datum/port/input/option/list_options
+
+	var/datum/port/input/input_list
+	var/datum/port/input/user
+	var/datum/port/input/input_name
+
+
+	var/datum/port/output/output
+	var/datum/port/output/success
+	var/datum/port/output/failure
```

Comment:
[Autodoc](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md) please

Reply:
Let me know if these comments are clear enough to other people